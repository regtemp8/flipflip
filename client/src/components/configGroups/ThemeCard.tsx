import * as React from 'react'
import { cx } from '@emotion/css'

import { type Theme, Typography } from '@mui/material'

import { makeStyles } from 'tss-react/mui'

import ThemeColorPicker from '../config/ThemeColorPicker'

import BaseSwitch from '../common/BaseSwitch'
import {
  setThemeMode,
  setThemePalettePrimary,
  setThemePaletteSecondary
} from '../../store/app/slice'
import {
  selectAppThemeMode,
  selectAppThemePalettePrimaryMain,
  selectAppThemePaletteSecondaryMain
} from '../../store/app/selectors'

const useStyles = makeStyles()((theme: Theme) => ({
  themePicker: {
    [theme.breakpoints.up('sm')]: {
      maxWidth: theme.spacing(47)
    }
  },
  gutterBottom: {
    marginBottom: theme.spacing(2)
  }
}))

function ThemeCard() {
  const { classes } = useStyles()
  return (
    <React.Fragment>
      <div>
        <BaseSwitch
          label="Dark Mode"
          selector={selectAppThemeMode()}
          action={setThemeMode}
        />
      </div>
      <div className={cx(classes.themePicker, classes.gutterBottom)}>
        <Typography>Primary Color</Typography>
        <ThemeColorPicker
          selector={selectAppThemePalettePrimaryMain()}
          action={setThemePalettePrimary}
        />
      </div>
      <div className={classes.themePicker}>
        <Typography>Secondary Color</Typography>
        <ThemeColorPicker
          selector={selectAppThemePaletteSecondaryMain()}
          action={setThemePaletteSecondary}
        />
      </div>
    </React.Fragment>
  )
}

;(ThemeCard as any).displayName = 'ThemeCard'
export default ThemeCard
