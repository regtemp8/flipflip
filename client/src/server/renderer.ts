import createCache, { type EmotionCache } from '@emotion/cache'
import { configureStore } from '@reduxjs/toolkit'
import { AppStorage, SystemConstants } from 'flipflip-common'
import createAppReducer from '../store/app/slice'
import createAudioReducer from '../store/audio/slice'
import createCaptionScriptReducer from '../store/captionScript/slice'
import createClipReducer from '../store/clip/slice'
import createLibrarySourceReducer from '../store/librarySource/slice'
import createPlaylistReducer from '../store/playlist/slice'
import createSceneReducer from '../store/scene/slice'
import createSceneGroupReducer from '../store/sceneGroup/slice'
import createTagReducer from '../store/tag/slice'
import playersSlice from '../store/player/slice'
import videoClipperSlice from '../store/videoClipper/slice'
import scenePickerSlice from '../store/scenePicker/slice'
import captionScriptorSlice from '../store/captionScriptor/slice'
import sceneDetailSlice from '../store/sceneDetail/slice'
import audioLibrarySlice from '../store/audioLibrary/slice'
import sourceScraperSlice from '../store/sourceScraper/slice'
import createConstantsReducer from '../store/constants/slice'
import { fromAppStorage } from '../store/app/convert'
import createDisplayReducer from '../store/display/slice'
import createDisplayViewReducer from '../store/displayView/slice'
import createScenePlaylistItemReducer from '../store/scenePlaylistItem/slice'
import createDisplayPlaylistItemReducer from '../store/displayPlaylistItem/slice'

export function createEmotionCache(): EmotionCache {
  return createCache({ key: 'css' })
}

export function createReduxStore(
  appStorage?: AppStorage,
  constants?: SystemConstants
) {
  const state = appStorage != null ? fromAppStorage(appStorage) : undefined
  return configureStore({
    reducer: {
      app: createAppReducer(state?.app),
      audio: createAudioReducer(state?.audio),
      captionScript: createCaptionScriptReducer(state?.captionScript),
      clip: createClipReducer(state?.clip),
      librarySource: createLibrarySourceReducer(state?.librarySource),
      playlist: createPlaylistReducer(state?.playlist),
      displayPlaylistItem: createDisplayPlaylistItemReducer(
        state?.displayPlaylistItem
      ),
      scenePlaylistItem: createScenePlaylistItemReducer(
        state?.scenePlaylistItem
      ),
      scene: createSceneReducer(state?.scene),
      sceneGroup: createSceneGroupReducer(state?.sceneGroup),
      display: createDisplayReducer(state?.display),
      displayView: createDisplayViewReducer(state?.displayView),
      tag: createTagReducer(state?.tag),
      constants: createConstantsReducer(constants),
      // components
      players: playersSlice,
      videoClipper: videoClipperSlice,
      scenePicker: scenePickerSlice,
      captionScriptor: captionScriptorSlice,
      sceneDetail: sceneDetailSlice,
      audioLibrary: audioLibrarySlice,
      sourceScraper: sourceScraperSlice
    }
  })
}
