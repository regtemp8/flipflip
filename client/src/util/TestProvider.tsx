import * as React from 'react'
import { Provider } from 'react-redux'
import { createTheme } from '@mui/material'
import { StyledEngineProvider, ThemeProvider } from '@mui/material/styles'
import defaultTheme from 'flipflip-common'

export default class TestProvider extends React.Component {
  readonly props: {
    children: any
    store: any
  }

  disableTransitions = {
    defaultProps: {
      disablePortal: true,
      hideBackdrop: true,
      TransitionComponent: ({ children }) => children
    }
  }

  render() {
    const theme = createTheme({
      ...defaultTheme,
      // disable Transitions when runing tests, throws errors otherwise.
      components: {
        MuiMenu: this.disableTransitions,
        MuiPopover: this.disableTransitions,
        MuiDialog: this.disableTransitions,
        MuiModal: this.disableTransitions
      }
    } as any)
    return (
      <Provider store={this.props.store}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={theme}>{this.props.children}</ThemeProvider>
        </StyledEngineProvider>
      </Provider>
    )
  }
}
