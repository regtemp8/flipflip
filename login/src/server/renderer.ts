import createCache, { type EmotionCache } from "@emotion/cache";

export function createEmotionCache(nonce?: string): EmotionCache {
  return createCache({ key: "css", nonce });
}
