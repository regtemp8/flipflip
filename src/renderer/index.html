<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>FlipFlip</title>
    <!-- Responsive meta tag -- ensures proper rendering and touch zooming for all devices -->
    <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width, shrink-to-fit=no" />
    <!-- Roboto Font -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap" />
    <!-- Font Icons -->
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons" />

    <meta
      http-equiv="Content-Security-Policy"
      content="frame-src https://www.imagefap.com http://www.imagefap.com https://hypno.nimja.com/ http://hypno.nimja.com/; font-src 'self' https://fonts.gstatic.com; style-src 'self' https://fonts.googleapis.com 'unsafe-inline'"
    />
  </head>

  <body>
    <div id="app" class="app"></div>
  </body>
</html>
