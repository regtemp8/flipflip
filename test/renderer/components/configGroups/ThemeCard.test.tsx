import React from "react";
import { describe, it, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import ThemeCard from "../../../../src/renderer/components/configGroups/ThemeCard";
import TestProvider from "../../../util/TestProvider";
import { createTheme } from "@mui/material";
import defaultTheme from "../../../../src/renderer/data/theme";

jest.mock('../../../../src/renderer/components/config/ThemeColorPicker', () => 'ThemeColorPicker');

describe("ThemeCard", () => {
  it("should match snapshot", () => {
    const component = renderer.create(
      <TestProvider>
        <ThemeCard
          theme={createTheme(defaultTheme as any)}
          onChangeThemeColor={(colorTheme, primary) => {}}
          onToggleDarkMode={() => {}}
        />
      </TestProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
