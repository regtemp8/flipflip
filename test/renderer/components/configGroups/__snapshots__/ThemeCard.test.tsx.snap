// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeCard should match snapshot 1`] = `
[
  <div>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-1flv81n-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Dark Mode
      </span>
    </label>
  </div>,
  <div
    className="ThemeCard-themePicker-1 ThemeCard-gutterBottom-2"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Primary Color
    </p>
    <ThemeColorPicker
      currentColor="#3f51b5"
      onChangeColor={[Function]}
    />
  </div>,
  <div
    className="ThemeCard-themePicker-1"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Secondary Color
    </p>
    <ThemeColorPicker
      currentColor="#f50057"
      onChangeColor={[Function]}
    />
  </div>,
]
`;
